server/server.js

```diff
@@ -254,48 +240,50 @@ app.post('/upload', (req, res, next) => {
   log.info('meta', meta);
   req.pipe(req.busboy);
 
-  req.busboy.on('file', (fieldname, file, filename) => {
+  req.busboy.on('file', async (fieldname, file, filename) => {
     log.info('Uploading:', newId);
 
-    storage.set(newId, file, filename, meta).then(
-      () => {
-        const protocol = conf.env === 'production' ? 'https' : req.protocol;
-        const url = `${protocol}://${req.get('host')}/download/${newId}/`;
-        res.json({
-          url,
-          delete: meta.delete,
-          id: newId
-        });
-      },
-      err => {
-        if (err.message === 'limit') {
-          return res.sendStatus(413);
-        }
-        res.sendStatus(500);
+    try {
+      await storage.set(newId, file, filename, meta);
+
+      const protocol = conf.env === 'production' ? 'https' : req.protocol;
+      const url = `${protocol}://${req.get('host')}/download/${newId}/`;
+      res.json({
+        url,
+        delete: meta.delete,
+        id: newId
+      });
+    } catch (e) {
+      if (e.message === 'limit') {
+        return res.sendStatus(413);
       }
-    );
+      res.sendStatus(500);
+    }
   });
 
-  req.on('close', err => {
-    storage
-      .forceDelete(newId)
-      .then(err => {
-        if (!err) {
-          log.info('Deleted:', newId);
-        }
-      })
-      .catch(err => {
-        log.info('DeleteError:', newId);
-      });
+  req.on('close', async err => {
```

Comment:
Unrelated to PR, but I just noticed that the `.on('close')` may return an `err` response here (apparently), which we aren't checking, but then we're overwriting the `err` value below on line 266.  🤷‍♀️ 
Behavior looks the same as it was previously, but there may be dragons here. Very small, uninteresting dragons.